### PR TITLE
fix: remove grpc-client-nginx-module

### PR DESCRIPTION
**Pull request by sheharyaar**
In APISIX, a Golang library was previously used to effectively implement a gRPC client and bind it to the Nginx ecosystem. It is called [grpc-client-nginx-module](https://github.com/api7/grpc-client-nginx-module). APISIX no longer uses this, it was removed in the PR https://github.com/apache/apisix/pull/9456. This PR removes it from the build-tools

### CODE REVIEW

File: build-apisix-base.sh [hunk 1]
```diff
@@ -112,8 +102,6 @@ ld_opt=${ld_opt:-}
 luajit_xcflags=${luajit_xcflags:="-DLUAJIT_NUMMODE=2 -DLUAJIT_ENABLE_LUA52COMPAT"}
 no_pool_patch=${no_pool_patch:-}
 # TODO: remove old NGX_HTTP_GRPC_CLI_ENGINE_PATH once we have released a new
```
Comment (starsz) [inline build-apisix-base.sh:104]: I think we can remove this line.

**Review by starsz — COMMENTED** [review]
Could write more details in the description ?
In the PR of APISIX (https://github.com/api7/apisix-build-tools/pull/370) we removed the etcd.use_grpc feature, so we don't need this module.

By the way, ci failed.

**Comment by sheharyaar** [comment]
> By the way, ci failed.

For all the CI, this step of updating and installing tools failed :

image

**Comment by monkeyDluffy6017** [comment]
We should make the ci pass

File: Makefile [hunk 2]
```diff
@@ -98,6 +101,7 @@ define build_runtime
 		--build-arg IMAGE_BASE=$(image_base) \
 		--build-arg IMAGE_TAG=$(image_tag) \
 		--build-arg CODE_PATH=$(4) \
+		--build-arg NPM_CONFIG_STRICT_SSL=false \
```
Comment (starsz) [inline Makefile:104]: It's a build-arg, and we don't define this in the dockerfile.
I think you can turn off the npm SSL check before the `make build`.
refer: https://github.com/api7/apisix-build-tools/blob/master/utils/install-common.sh#L161

Comment (starsz) [reply Makefile:104]: I create a PR to fix it.
refer: https://github.com/api7/apisix-build-tools/pull/373

**Review by starsz — APPROVED** [review]

**Pull request merged**